src/coreclr/vm/threads.cpp

```diff
@@ -6047,11 +6047,13 @@ BOOL Thread::SetStackLimits(SetStackLimitScope scope)
     {
         m_CacheStackBase  = GetStackUpperBound();
         m_CacheStackLimit = GetStackLowerBound();
+#ifndef TARGET_WASM // stack can start at address 0 on wasm/emscripten and usually does
```

Comment:
Is this still needed now that we are using `emscripten_stack_get_base`?

Reply:
Yeah, the stack limit is still at 0 in debug configuration. I made the condition reflect that, it probably doesn't change much as the assert is probably disabled in non-debug configurations anyway?

We can try to put something at the beginning of the memory. Not sure it is worth it. It shouldn't cause trouble, we would only reach that just before the C runtime stack overflow.